nubison_model/Service.py

```diff
@@ -19,48 +19,39 @@
 from nubison_model.utils import temporary_cwd
 
 
-def load_nubison_model(
-    mlflow_tracking_uri,
-    mlflow_model_uri,
-    prepare_artifacts: bool = False,
-):
+def load_nubison_mlflow_model(mlflow_tracking_uri, mlflow_model_uri):
+    if not mlflow_tracking_uri:
+        raise RuntimeError("MLflow tracking URI is not set")
+    if not mlflow_model_uri:
+        raise RuntimeError("MLflow model URI is not set")
 
     try:
-        if not mlflow_tracking_uri:
-            raise RuntimeError("MLflow tracking URI is not set")
-        if not mlflow_model_uri:
-            raise RuntimeError("MLflow model URI is not set")
         set_tracking_uri(mlflow_tracking_uri)
         mlflow_model = load_model(model_uri=mlflow_model_uri)
-
         nubison_mlflow_model = cast(
             NubisonMLFlowModel, mlflow_model.unwrap_python_model()
         )
-        if prepare_artifacts:
-            nubison_mlflow_model.prepare_artifacts()
-
-        # Get the NubisonModel instance from the MLflow model
-        nubison_model = nubison_mlflow_model.get_nubison_model()
     except Exception as e:
         raise RuntimeError(
             f"Error loading model(uri: {mlflow_model_uri}) from model registry(uri: {mlflow_tracking_uri})"
         ) from e
 
-    return nubison_model
+    return nubison_mlflow_model
 
 
 @contextmanager
 def test_client(model_uri):
-    app = build_inference_service(mlflow_model_uri=model_uri)
-    # Disable metrics for testing. Avoids Prometheus client duplicated registration error
-    app.config["metrics"] = {"enabled": False}
 
     # Create a temporary directory and set it as the current working directory to run tests
     # To avoid model initialization conflicts with the current directory
     test_dir = TemporaryDirectory()
-    with temporary_cwd(test_dir.name), TestClient(app.to_asgi()) as client:
```

Comment:
I'm curious about the role of TestClient(app.to_asgi()) as client previously and why there were initialization conflicts before

Reply:
It was a bug related to incorrect initialization timing. The artifact was downloaded in the current directory, then moved to a temporary directory to run the inference code. This process resulted in missing artifacts.